File: Backend/chat_real_time/src/main/java/ec/edu/espe/chat_real_time/Service/message/MessageService.java
package ec.edu.espe.chat_real_time.Service.message;

import ec.edu.espe.chat_real_time.dto.request.SendMessageRequest;
import ec.edu.espe.chat_real_time.dto.response.MessageResponse;
import ec.edu.espe.chat_real_time.model.user.User;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface MessageService {
  MessageResponse sendTextMessage(SendMessageRequest request, User user);
  MessageResponse sendFileMessage(Long roomId, String content, MultipartFile file, User user);
  List<MessageResponse> getRoomMessages(Long roomId, User user);
  MessageResponse getMessageById(Long messageId);
  void deleteMessage(Long messageId, User user);
}

File: frontend/viva-chat-ui/src/services/messageService.ts
import apiClient from '@/config/api';
import {
  ApiResponse,
  MessageResponse,
  SendMessageRequest,
} from '@/types/api';

export const messageService = {
  sendTextMessage: async (messageData: SendMessageRequest): Promise<ApiResponse<MessageResponse>> => {
    const response = await apiClient.post<ApiResponse<MessageResponse>>(
      '/messages/text',
      messageData
    );
    return response.data;
  },

  sendFileMessage: async (
    roomId: number,
    content: string,
    file: File
  ): Promise<ApiResponse<MessageResponse>> => {
    const formData = new FormData();
    formData.append('roomId', roomId.toString());
    formData.append('content', content);
    formData.append('file', file);

    const response = await apiClient.post<ApiResponse<MessageResponse>>(
      '/messages/file',
      formData,
      {
        headers: {
          'Content-Type': 'multipart/form-data',
        },
      }
    );
    return response.data;
  },

  getRoomMessages: async (roomId: number): Promise<ApiResponse<MessageResponse[]>> => {
    const response = await apiClient.get<ApiResponse<MessageResponse[]>>(
      `/messages/room/${roomId}`
    );
    return response.data;
  },

  getMessage: async (messageId: number): Promise<ApiResponse<MessageResponse>> => {
    const response = await apiClient.get<ApiResponse<MessageResponse>>(
      `/messages/${messageId}`
    );
    return response.data;
  },

  deleteMessage: async (messageId: number): Promise<ApiResponse<null>> => {
    const response = await apiClient.delete<ApiResponse<null>>(
      `/messages/${messageId}`
    );
    return response.data;
  },
};
